src/library_fs.js

```diff
@@ -1301,7 +1301,7 @@ mergeInto(LibraryManager.library, {
 #endif // ENVIRONMENT_MAY_BE_NODE
       } else {
         // default for ES5 platforms
-        random_device = function() { return (Math.random()*256)|0; };
+        random_device = function() { throw "emscripten random_device implementation must be run on a modern browser to support safe random number generation."; };
```

Comment:
Please use `abort("..text..")` instead of throw - abort will also mark the program as halted, so an `emscripten_main_loop` etc. will know to stop itself next time it is called.

For the error text, I'd prefer something short, just to not increase code size too much. How about just `"random_device"`, and in a comment (which in an optimized build would be removed) to give the extra detail?